### PR TITLE
chore: implemented a json object for greeting hello world

### PR DESCRIPTION
**Pull request by Odelola**
This is the picture of the file
backend-api-endpoint-2

The main challenges I faced were trying to configure the project for the IDE as Maven.
Resolving errors made with the dependency installation.
Remebering to add SemiColon, remebering your annotations and remembering to import packages especially Java native ones.

### CODE REVIEW

File: src/main/java/com/example/Application.java
Comment (adeloyedeji) [inline src/main/java/com/example/Application.java:1]: This file is not needed. There is an already existing file with @SpringBootApplication.

@Springboot application annotation is what is used to identify and bootstrap the starting point of the springboot application. This is already defined in `HelloAppication.java`

File: src/main/java/com/example/hello/HelloApplication.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 @SpringBootApplication
 public class HelloApplication {
 
-	public static void main(String[] args) {
+	public static void Hellomain(String[] args) {
```
Comment (adeloyedeji) [inline src/main/java/com/example/hello/HelloApplication.java]: Revert changes to this file and leave as main, this is the main startup file

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 spring.application.name=hello
-spring.datasource.url=
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-spring.datasource.username=
-spring.datasource.password=
\ No newline at end of file
+spring.datasource.url=jdbc:mysql://localhost:3306/spotflow_payment_gateway?createDatabaseIfNotExist=true
```
Comment (adeloyedeji) [inline src/main/resources/application.properties]: like I said, you don't commit secrets to application.properties file. Remove username and password here

Comment (adeloyedeji) [reply src/main/resources/application.properties]: including the jdbc url

File: src/test/java/com/example/hello/HelloApplicationTests.java [hunk 3]
```diff
@@ -6,8 +6,9 @@
 @SpringBootTest
 class HelloApplicationTests {
 
-	@Test
```
Comment (adeloyedeji) [inline src/test/java/com/example/hello/HelloApplicationTests.java:9]: there was no significant change here, why the edit?

File: src/main/java/com/example/hello/hello/controller/HelloController.java [hunk 4]
```diff
@@ -1,16 +1,24 @@
...
+
 
 
 @RestController
 @RequestMapping("api/v1/hello")
 public class HelloController {
     @GetMapping
-    public String helloString() {
-        return "Hello World";
+    @ResponseBody
```
Comment (adeloyedeji) [inline src/main/java/com/example/hello/hello/controller/HelloController.java:16]: why did you remove the `@GetMapping` annotation?

Comment (adeloyedeji) [reply src/main/java/com/example/hello/hello/controller/HelloController.java:16]: Always make sure to follow already existing pattern when you are working on a code base, never re-invent the wheel until you know what is happening in and out of the system.

**Comment by adeloyedeji** [comment]
Also, focus on good development and version control practices: the title of PR does not need to include chore:

The name of your branch also doesn't match what is in the PR, you didn't implement anything related to onboarding, let your branch name always be relatable to the actual code in the PR.

**Review by adeloyedeji — CHANGES_REQUESTED** [review]
Review the comments added and update the PR.

File: src/main/java/com/example/hello/hello/controller/HelloController.java (hunk 4 above)
Comment (Odelola) [reply src/main/java/com/example/hello/hello/controller/HelloController.java:16]: Alright sir The @GetMapping is still there

File: src/test/java/com/example/hello/HelloApplicationTests.java (hunk 3 above)
Comment (Odelola) [reply src/test/java/com/example/hello/HelloApplicationTests.java:9]: The test prevented me from installing dependecies

File: src/main/java/com/example/hello/HelloApplication.java (hunk 1 above)
Comment (Odelola) [reply src/main/java/com/example/hello/HelloApplication.java]: I have changed this file back to main
And I deleted the Application File I kept there before

File: src/main/java/com/example/Application.java
Comment (Odelola) [reply src/main/java/com/example/Application.java:1]: I have deleted the file and resolved it

File: src/test/java/com/example/hello/HelloApplicationTests.java (hunk 3 above)
Comment (adeloyedeji) [reply src/test/java/com/example/hello/HelloApplicationTests.java:9]: No, it can't prevent you from adding  dependencies

**Review by adeloyedeji — APPROVED** [review]

**Comment by adeloyedeji** [comment]
PR has been approved, you can merge it now
